src/runtime_src/core/edge/user/shim.h

```diff
@@ -156,7 +157,7 @@ class shim {
   int xclLog(xrtLogMsgLevel level, const char* tag, const char* format, ...);
 
 #ifdef XRT_ENABLE_AIE
-  zynqaie::Aie *aieArray;
+  std::shared_ptr<zynqaie::Aie> aieArray;
```

Comment:
What the lifetime of this object?  Is really shared ownership?  Or should it be owned by the shim object and die when the shim object is deleted?  I think it's unique ownership in which case it should be std::unique_ptr and getAieArray() should return the raw pointer.   This doesn't change functionality, just communicates intent.

Reply:
Thanks Soren. With the returning raw pointer suggestion, I can make getAieArray() works with unique pointer.